multiqc/modules/dedup/dedup.py

```diff
@@ -50,42 +52,38 @@ def __init__(self):
             plot = self.dedup_alignment_plot()
         )
 
-
-    def parse_dedup_log(self, f):
-        regexes = {
-            'input_file': r"Input file:\s+(\S+)",
-            'total_reads': r"Total reads:\s+(\d+)",
-            'reverse_removed': r"Reverse removed:\s+(\d+)",
-            'forward_removed': r"Forward removed:\s+(\d+)",
-            'merged_removed': r"Merged removed:\s+(\d+)",
-            'total_removed': r"Total removed:\s+(\d+)",
-            'duplication_rate': r"Duplication Rate:\s+([\d\.]+)",
-        }
-        parsed_data = dict()
-        for k, r in regexes.items():
-            r_search = re.search(r, f['f'], re.MULTILINE)
-            if r_search:
-                try:
-                    parsed_data[k] = float(r_search.group(1))
-                except ValueError:
-                    parsed_data[k] = r_search.group(1)
+    #Parse our nice little JSON file
+    def parseJSON(self, f):
+        """ Parse the JSON output from DeDup and save the summary statistics """
         try:
-            parsed_data['not_removed'] = parsed_data['total_reads'] - parsed_data['reverse_removed'] - parsed_data['forward_removed'] - parsed_data['merged_removed']
-        except KeyError:
-            log.debug('Could not calculate "not_removed"')
+            parsed_json = json.load(f['f'])
+        except Exception as e:
+            print(e)
+            log.warn("Could not parse DeDup JSON: '{}'".format(f['fn']))
+            return None
+
+        #Get sample name from JSON first
+        s_name = self.clean_s_name(parsed_json['metadata']['sample_name'],'')
```

Comment:
What happens if someone has a valid JSON file that is called `something_dedup.json` that is from a different origin? Given that the filename pattern isn't _super_ tool specific, I think it would be good to add in an extra check for a few expected keys first. Then we can throw a debug message and `return None` before getting here where we'd have a `KeyError` exception..